flask_server.py

```diff
@@ -110,6 +84,57 @@ def logs():
 
 ################### API ROUTES ###################
 
+def get_index_settings(user_settings):
+        if scraper_runner.args:
+            loop_mode = not scraper_runner.shutting_down
+            media_mode = scraper_runner.args['media_mode']
+            ids_source = scraper_runner.args['ids_source']
+        else:
+            loop_mode = False
+            media_mode = "all"
+            ids_source = "all"
```

Comment:
You should be able to replace it with 3 lines of conditional expressions https://stackoverflow.com/a/22983107